libflux/src/semantic/tests.rs

```diff
@@ -255,715 +255,2377 @@ fn literals() {
-        //                "b" => "forall [] t1",
-        //            ],
-        //            src: &src,
-        //            exp: map![
-        //                "a" => "forall [] string",
-        //                "b" => "forall [] regexp",
-        //                "c" => "forall [] bool",
-        //            ],
-        //        }
-        test_infer! {
-            env: map![
-                "a" => "forall [] string",
-                "b" => "forall [] regexp",
-            ],
-            src: &src,
-            exp: map![
-                "c" => "forall [] bool",
-            ],
-        }
-        test_infer_err! {
-            env: map![
-                "a" => "forall [] float",
-                "b" => "forall [] regexp",
-            ],
-            src: &src,
-        }
-        test_infer_err! {
-            env: map![
-                "a" => "forall [] string",
-                "b" => "forall [] float",
-            ],
-            src: &src,
-        }
-        test_infer_err! {
-            env: map![
-                "a" => "forall [] regexp",
-                "b" => "forall [] string",
-            ],
-            src: &src,
-        }
-        test_infer_err! {
-            env: map![
-                "a" => "forall [] float",
-                "b" => "forall [] int",
-            ],
-            src: &src,
-        }
-        //        test_infer! {
-        //            env: map![
-        //                "a" => "forall [] t0",
-        //                "b" => "forall [] regexp",
-        //            ],
-        //            src: &src,
-        //            exp: map![
-        //                "a" => "forall [] string",
-        //                "c" => "forall [] bool",
-        //            ],
-        //        }
-        //        test_infer! {
-        //            env: map![
-        //                "a" => "forall [] string",
-        //                "b" => "forall [] t0",
-        //            ],
-        //            src: &src,
-        //            exp: map![
-        //                "b" => "forall [] regexp",
-        //                "c" => "forall [] bool",
-        //            ],
-        //        }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] regexp",
+            "b" => "forall [] regexp",
+        ],
+        src: r#"
+            a + b
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] {a: int | b: float}",
+            "b" => "forall [] {a: int | b: float}",
+        ],
+        src: r#"
+            a + b
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] [int]",
+            "b" => "forall [] [int]",
+        ],
+        src: r#"
+            a + b
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] float",
+            "b" => "forall [] int",
+        ],
+        src: r#"
+            c = a + b
+        "#,
```

Comment:
this will drive users crazy ahahahah. Cannot sum int and float. Think we need a `Number` kind for which addition is defined.

Reply:
You are absolutely right. The plan is to add such a kind so that something like this would be supported.